lightning/src/offers/signer.rs

```diff
@@ -24,6 +25,8 @@ use crate::prelude::*;
 
 const DERIVED_METADATA_HMAC_INPUT: &[u8; 16] = &[1; 16];
 const DERIVED_METADATA_AND_KEYS_HMAC_INPUT: &[u8; 16] = &[2; 16];
+const WITHOUT_ENCRYPTED_PAYMENT_ID_HMAC_INPUT: &[u8; 16] = &[3; 16];
+const WITH_ENCRYPTED_PAYMENT_ID_HMAC_INPUT: &[u8; 16] = &[4; 16];
```

Comment:
Could you add docs for the reasoning behind inputting these consts? It seems related to this discussion: https://github.com/lightningdevkit/rust-lightning/pull/1989#discussion_r1170466651